src/x86_64/apic.c

```diff
@@ -96,7 +96,7 @@ void msi_format(u32 *address, u32 *data, int vector)
     u32 dm = 0;             // destination mode: ignored if rh == 0
     u32 rh = 0;             // redirection hint: 0 - disabled
     u32 destination = 0;    // destination APIC
-    *address = (0xfee << 20) | (destination << 12) | (rh << 3) | (dm << 2);
```

Comment:
This is probably too nitpicky, but if we wanted to keep the shift value separate from the operand, would the ULL suffix on the 0xfee constant eliminate the UBSan error?

Reply:
Good point, indeed specifying the 0xfee constant as a signed number (as in `0xfeeu`) eliminates the UBSan error, which makes total sense. I updated the code accordingly.